### ISSUE TITLE
Suboptimal compact registers display

### ISSUE THREAD

**Issue by disconnect3d** [pwndbg/pwndbg#1202]
When we display compact registers I guess the order in which we display regs may be suboptimal?

![image](https://user-images.githubusercontent.com/10009354/193497707-ca990add-e00f-42b6-a32e-3d91fb52e842.png)

I think less important registers like r8-r15 should be on the right side.

Also: should we allow for specifying the orders in which registers are displayed? I am not sure about that.

**Comment by gsingh93** [comment]
Fixing this is a bit trickier than I first thought. If the output from displaying a register in the second column can't fit in the column, it gets put on the next line. This means that even if we try to order the registers in a particular way, something in the second column could end up in the first column if the line is too long.

**Comment by disconnect3d** [comment]
Wait, does this mean that if register values change between two breakpoints, you may get differently ordered regs in the two context displays? That's not great :<

**Comment by gsingh93** [comment]
Yes, exactly. This happened to me today, it's not a great experience. But I'm also concerned about truncating the displayed register value, so I'm not sure what to do here.

**Comment by disconnect3d** [comment]
We should just enforce the same order of regs and apply some limit for telescoping the registers or just limiting their length.

**Comment by disconnect3d** [comment]
Honestly, I don't know if anyone is using compact regs at this point. I used it few times just for screenshots/presentations or when I really needed more context. I think we can close this for now until someone really needs this.

Closing.

**Comment by k4lizen** [comment]
hehe i was just talking to a friend a couple of days ago how i wish this was fixed
